spec/views/report/_form_expression_buttons.html.haml_spec.rb

```diff
@@ -0,0 +1,35 @@
+describe 'report/_form_filter.html.haml' do
+  let(:report) { FactoryGirl.create(:miq_report) }
+  let(:expkey) { :record_filter }
+  let(:exp_table) { [["Example AFTER \"value1\"", 1]] }
+  let(:expression) { {"after" => {"field" => "field_example", "value" => "value1"}, :token => 1} }
+  let(:edit) { { :rpt_id => report.id, expkey => {:exp_table => exp_table, :expression => expression}} }
+
+  before do
+    assign(:expkey, expkey)
```

Comment:
The `expkey` variable is not being used anywhere else, so you can simplify it here to:
```
assign(:expkey, :record_filter)
```

Reply:
This is ok if I also change the line 6 in this spec test

Reply:
Hmm, that's confusing a little bit, I think it's better to use the `:record_filter` there statically too. Dynamically assigning hash keys is not a trivial thing and might confuse someone the same way it did with me.